src/power_grid_model/validation/rules.py

```diff
@@ -373,6 +373,9 @@ def none_match_comparison(  # pylint: disable=too-many-arguments
     if default_value_2 is not None:
         set_default_value(data=data, component=component, field=field, default_value=default_value_2)
     component_data = data[component]
+    if not isinstance(component_data, np.ndarray):
+        raise NotImplementedError()  # TODO(mgovers): add support for columnar data
```

Comment:
@figueroa1395 is this still necessary?

Reply:
For the same reason, all validation is done in row format, it shouldn't be either.

However, before merging, we can test by re-basing with main (which already has the PR adding the support for columnar validation), and I can briefly make a branch on top of this to do the validation upgrade, and then we are sure we don't miss anything. @mgovers @nitbharambe what do you think?

Reply:
let's leave it as is and remove it as a follow-up. it's fine. your original PR should already ensure you don't end up here, but leaving it at least makes it clear what is hit and when.